Sources/Workspace/Workspace.swift

```diff
@@ -1204,7 +1204,7 @@ extension Workspace {
     /// generated by SwiftPM.
     public func fileAffectsSwiftOrClangBuildSettings(filePath: AbsolutePath, packageGraph: ModulesGraph) -> Bool {
         // TODO: Implement a more sophisticated check that also verifies if the file is in the sources directories of the passed in `packageGraph`.
-        FileRuleDescription.builtinRules.contains { fileRuleDescription in
+        filePath.extension != "h" && FileRuleDescription.builtinRules.contains { fileRuleDescription in
```

Comment:
Would it be better to exclude all the extensions from `FileRuleDescription.header`?

Reply:
Good catch, thanks!